### PR TITLE
[SPARK-41576][SQL] Assign name to _LEGACY_ERROR_TEMP_2051

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to assign name to _LEGACY_ERROR_TEMP_2051, "DATA_SOURCE_NOT_FOUND".

### Why are the changes needed?

We should assign proper name to _LEGACY_ERROR_TEMP_*

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

`./build/sbt "sql/testOnly org.apache.spark.sql.SQLQueryTestSuite*`

### CODE REVIEW

**Review by MaxGekk — CHANGES_REQUESTED** [review]
@itholic Could you add a test for the error class, please.

**Comment by itholic** [comment]
Thanks! Migrated the existing test to use `checkError`.

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -426,6 +426,11 @@
     ],
     "sqlState" : "42000"
   },
+  "DATA_SOURCE_NOT_FOUND" : {
+    "message" : [
+      "Failed to find data source: <provider>. Please find packages at https://spark.apache.org/third-party-projects.html"
```
Comment (LuciferYang) [inline core/src/main/resources/error/error-classes.json:431]: hmmm.... add single quotation for the html url?

Comment (itholic) [reply core/src/main/resources/error/error-classes.json]: Just added single quotation around the URL, thanks!

File: core/src/main/resources/error/error-classes.json [hunk 2]
```diff
@@ -426,6 +426,11 @@
     ],
     "sqlState" : "42000"
   },
+  "DATA_SOURCE_NOT_FOUND" : {
+    "message" : [
+      "Failed to find data source: <provider>. Please find packages at `https://spark.apache.org/third-party-projects.html`"
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:431]: nit:
```suggestion
      "Failed to find the data source: <provider>. Please find packages at `https://spark.apache.org/third-party-projects.html`."
```

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala [hunk 3]
```diff
@@ -734,7 +734,7 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase {
   def failedToFindDataSourceError(
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala:734]: Let's make it consistent w/ the error class name:
```suggestion
  def dataSourceNotFoundError(
```

Comment (itholic) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala]: Thanks!

**Review by MaxGekk — APPROVED** [review]
Waiting for CI.

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @itholic.

**Pull request closed without merging**